aten/src/ATen/native/mkl/SpectralOps.cpp

```diff
@@ -479,6 +479,16 @@ static Tensor& _exec_fft(Tensor& out, const Tensor& self, IntArrayRef out_sizes,
   const auto value_type = c10::toRealValueType(input.scalar_type());
   out.resize_(batched_out_sizes, MemoryFormat::Contiguous);
 
+  // fix mkl issue
+  // https://github.com/pytorch/pytorch/issues/154477
+  auto istrides = input.strides();
+  for (const auto& stride : istrides) {
+    if (stride == 0) {
+      input = input.clone(MemoryFormat::Contiguous);
+      break;
+    }
+  }
```

Comment:
do you mind submitting a separate pr? And why just zero stride? (and i bet there are already a helper function that checks for something like that)
Also it would be nice to reference MKL bug report or some reference to the doc explaining why this is no longer valid...